core/util.py

```diff
@@ -0,0 +1,34 @@
+from time import sleep
+
+
+def retry(func) -> bool:
```

Comment:
There are libraries that provide configurable methods of retrying till success, eg. https://github.com/invl/retry
If we don't need the configurability of these libraries we can stick to this implementation.

Reply:
For the time being I don't think we need configurability.